coderd/prometheusmetrics/aggregator.go

```diff
@@ -64,15 +68,58 @@ type annotatedMetric struct {
 	expiryDate time.Time
 }
 
+type metricKey struct {
+	username      string
+	workspaceName string
+	agentName     string
+	templateName  string
+
+	metricName string
+	labelsStr  string
+}
+
+func (m1 metricKey) Equal(m2 metricKey) bool {
```

Comment:
now that it's a struct of strings, I think the ordinary `==` operator does the right thing and this is not needed.

Reply:
Removed `Equal` function.